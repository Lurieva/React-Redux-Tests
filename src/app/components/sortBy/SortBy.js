import React  from 'react';
import PropTypes from 'prop-types';

import { SORT_BY } from '../../app.config';
import { RadioGroup } from '../../shared';

import './sortBy.scss';

const SortBy = ({ sortBy, onChangeSortBy }) => (
    <RadioGroup name={'sortBy'}
                groupClassName={'sortBy'}
                param={sortBy} 
                options={[SORT_BY.RELEASE_DATE, SORT_BY.RATING]}
                title={'Sort By'}
                onChange={onChangeSortBy} />
);

SortBy.propTypes = {
    sortBy: PropTypes.string.isRequired,
    onChangeSortBy: PropTypes.func.isRequired
};

export default SortBy;
