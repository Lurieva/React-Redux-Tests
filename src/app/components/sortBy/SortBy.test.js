import React from 'react';
import { shallow } from 'enzyme';
import SortBy from './SortBy';

jest.mock('../../shared/RadioGroup/RadioGroup', () => 'RadioGroup');

describe('SortBy', () => {
    const mockFn = jest.fn();

    it('should be defined', () => {
        expect(SortBy).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<SortBy sortBy='sort'
                                     onChangeSortBy={mockFn} />);
        expect(tree).toMatchSnapshot();
    });
});