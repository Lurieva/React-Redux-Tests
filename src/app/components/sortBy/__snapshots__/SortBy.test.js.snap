// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SortBy
    onChangeSortBy={[MockFunction]}
    sortBy="sort"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "groupClassName": "sortBy",
      "name": "sortBy",
      "onChange": [MockFunction],
      "options": Array [
        Object {
          "label": "release date",
          "value": "release_date",
        },
        Object {
          "label": "rating",
          "value": "vote_average",
        },
      ],
      "param": "sort",
      "title": "Sort By",
    },
    "ref": null,
    "rendered": null,
    "type": "RadioGroup",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "groupClassName": "sortBy",
        "name": "sortBy",
        "onChange": [MockFunction],
        "options": Array [
          Object {
            "label": "release date",
            "value": "release_date",
          },
          Object {
            "label": "rating",
            "value": "vote_average",
          },
        ],
        "param": "sort",
        "title": "Sort By",
      },
      "ref": null,
      "rendered": null,
      "type": "RadioGroup",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
